File: 2692-take-gifts-from-the-richest-pile/take-gifts-from-the-richest-pile.java
class Solution {
    public long pickGifts(int[] gifts, int k) {
        long result = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<Integer>(Collections.reverseOrder());
        for(int i: gifts)
            pq.add(i);
        while (k > 0){
            int x = pq.poll();
            pq.add((int)Math.floor(Math.sqrt(x)));
            k--;
        }
        while (pq.size() > 0){
            int temp = pq.poll();
            System.out.println(temp);
            result += temp;
            
        }

        return result;
    }
}

File: 2692-take-gifts-from-the-richest-pile/take-gifts-from-the-richest-pile.cpp
class Solution {
public:
    long long pickGifts(vector<int>& gifts, int k) {
        long long result = 0;
        priority_queue<int> pq;
        for(int i: gifts)
            pq.push(i);
        while (k > 0){
            int x = pq.top();
            pq.pop();
            pq.push(floor(sqrt(x)));
            k--;
        }
        while (pq.size() > 0)
        {
            result += pq.top();
            pq.pop();
        }
        return result;
    }
};